src/libostree/ostree-repo-commit.c

```diff
@@ -3286,22 +3286,33 @@ _ostree_repo_commit_modifier_apply (OstreeRepo               *self,
                                     GFileInfo                *file_info,
                                     GFileInfo               **out_modified_info)
 {
+  gboolean canonicalize_perms = FALSE;
+  gboolean has_filter = FALSE;
   OstreeRepoCommitFilterResult result = OSTREE_REPO_COMMIT_FILTER_ALLOW;
   GFileInfo *modified_info;
 
-  if (modifier == NULL ||
-      (modifier->filter == NULL &&
-       (modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) == 0))
+  /* Auto-detect bare-user-only repo, force canonical permissions. */
+  // XXX(lucab): This seems to break `checkout --union-identical`.
```

Comment:
Still need to investigate what's going on here. From what I've observed so far, the is-identical logic is failing due to a checksum mismatch on symlinks.

Reply:
Found the problem, fixed by https://github.com/ostreedev/ostree/pull/2415.